### PR TITLE
Use TMPDIR when testing if possible in tests to get round the opam sandbox

### PR DESCRIPTION
**Pull request by psafont**
for xapi_db_upgrade this means having to select one of three directories

This changes are needed for opam's 2.1.0 sanbox, which doesn't allow writing to /tmp

### CODE REVIEW

File: ocaml/tests/common/test_common.ml [hunk 1]
```diff
@@ -15,10 +15,10 @@
 open API
 
 (* A directory to use for temporary files. *)
-let working_area = "/tmp/xapi-test"
-
-(** Utility functions *)
-let id (x : 'a) : 'a = x
+let working_area =
+  Filename.concat
+    (Option.value ~default:"/tmp" (Sys.getenv_opt "TMPDIR"))
```
Comment (edwintorok) [inline ocaml/tests/common/test_common.ml:20]: Doesn't Filename module in OCaml have a way to read/use TMPDIR automatically? See `Filename.get_temp_dir_name`

Comment (edwintorok) [reply ocaml/tests/common/test_common.ml:20]: We could also use Filename.temp_file which also takes care of creating a unique temp name with a chosen suffix (although cleaning it up is up to the caller).

**Comment by psafont** [comment]
Now all tests pass on an xs-opam switch for all dependencies of xs-toolstack:

```
$ opam list -s --required-by xs-toolstack | xargs opam install -ty
```

File: ocaml/xapi/xapi_db_upgrade.ml [hunk 2]
```diff
@@ -303,7 +303,9 @@ let upgrade_bios_strings =
         let oem_manufacturer =
           let test_path =
             Option.map
-              (fun _ -> "/tmp/previousInventory")
+              (fun _ ->
```
Comment (robhoes) [inline ocaml/xapi/xapi_db_upgrade.ml:306]: Alternatively, we can just ditch this upgrade rule, which is for upgrades from XS 5.5, released on 2009.

**Review by robhoes — APPROVED** [review]

File: ocaml/xapi/xapi_db_upgrade.ml (hunk 2 above)
Comment (psafont) [reply ocaml/xapi/xapi_db_upgrade.ml:306]: I still would prefer to have this as we might have to backport this to yangtze

**Review by lindig — APPROVED** [review]

**Pull request merged**